packages/flutter/lib/src/services/text_input.dart

```diff
@@ -134,20 +135,201 @@ class TextInputType {
 }
 
 /// An action the user has requested the text input control to perform.
+///
+/// Each action represents a logical meaning, and also configures the soft
+/// keyboard to display a certain kind of action button. The visual appearance
+/// of the action button might differ between versions of the same OS.
+///
+/// Despite the logical meaning of each action, choosing a particular
+/// [TextInputAction] does not necessarily cause any specific behavior to
+/// happen. It is up to the developer to ensure that the behavior that occurs
+/// when an action button is pressed is appropriate for the action button chosen.
+///
+/// For example: If the user presses the keyboard action button on iOS when it
+/// reads "Emergency Call," the result should not be a focus change to the next
+/// TextField. This behavior is not logically appropriate for a button that says
+/// "Emergency Call."
```

Comment:
nit: we use the "programming" style of quotation in the docs, so the commas and periods should be after the quote marks.

Reply:
(this also applies in other parts of these docs)

Reply:
Yeah...gotta break this habit.